src/client_side_reply.cc

```diff
@@ -1656,6 +1656,8 @@ clientReplyContext::cloneReply()
 
     /* do header conversions */
     buildReplyHeader();
+
+    http->al->setReply(reply);
```

Comment:
This setReply() argument points to a to-be-sent response rather than the (adapted) received response, right? Is that what we want in ALE? Your new ALE::reply member documentation seems to imply that we want the _received_ (and possibly adapted) response instead. Please check and adjust this code or the new data member description.